src/components/pages/bc-umbrel.ts

```diff
@@ -0,0 +1,60 @@
+import {customElement} from 'lit/decorators.js';
+import {BitcoinConnectElement} from '../BitcoinConnectElement';
+import {withTwind} from '../twind/withTwind';
+import {html} from 'lit';
+import '../internal/bci-button';
+import {classes} from '../css/classes';
+import store from '../../state/store';
+import {webln} from '@getalby/sdk';
+
+@customElement('bc-umbrel')
+export class UmbrelPage extends withTwind()(BitcoinConnectElement) {
+  override render() {
+    return html`<div class="w-full">
+      <bc-navbar class="flex w-full" heading="Umbrel"></bc-navbar>
+      <div class="font-sans text-sm w-full">
+        <div class="px-8 pt-4 w-full">
+          <div class="mb-4 ${classes['text-neutral-secondary']}">
+            Install the App "NWC" from the Umbrel app store and click the
+            connect button below. If you don't use
+            <span class="italic">umbrel.local</span>
+            as your umbrel domain use the Generic NWC connector instead.
+          </div>
+          <bci-button @click=${this.onConnect}>
+            <span class="${classes['text-brand-mixed']}">Connect</span>
+          </bci-button>
+        </div>
+      </div>
+    </div>`;
+  }
+
+  private async onConnect() {
+    const nwc = webln.NostrWebLNProvider.withNewSecret({
+      authorizationUrl: 'http://umbrel.local:58000/apps/new',
+    });
+    await nwc.initNWC({
+      name: this._appName || 'Bitcoin Connect',
+    });
+
+    // FIXME: do not change route - extract skeleton loader and add loading state in this component
+    // and make connect throw an error on failure
+    store.getState().setRoute('/start');
+
+    await store.getState().connect({
+      nwcUrl: nwc.getNostrWalletConnectUrl(),
+      connectorName: 'Umbrel',
+      connectorType: 'nwc.umbrel',
+    });
+    if (!store.getState().connected) {
+      store.getState().setRoute('/nwc');
+      // TODO: show an error message directly on this page
+      alert('Failed to connect. Please check your NWC URL');
+    }
```

Comment:
Feels like quite a hack, doesn't it? Any ideas how we could simplify this?

Reply:
Yes, I think we can extract the skeleton loading code into a new component and render that from each page instead.

But each page will have a loading state and error messages - I think there will be quite a lot of duplication. I wonder if there is a way to simplify it as a follow up step.

I plan to address this in a separate PR. I made an issue here: https://github.com/getAlby/bitcoin-connect/issues/83

Reply:
Okay, cool! :sunglasses: